### ISSUE TITLE
[FEATURE REQUEST] Automatically check if SVGs contain raster images

### ISSUE THREAD

**Issue by Snailedlt** [devicons/devicon#1548]
### I have searched through the issues and didn't find my problem.

- [X] Confirm

### Problem

Currently raster images are allowed in the SVG's by the check-bot. This should not be the case, since raster images are not scalable, and therefore don't belong in our icons.

### Possible Solution

Add checks to `check-bot`. The bot should check for <img> and <image> tags

### Additional information

See #1517 for some examples and more info

**Comment by lunatic-fox** [comment]
Hello there! Perhaps, we can use this regular expression.
```regexp
/="data:image\/.+;base64,/gm
```

Well, let me try to convert it to Python: 😅
```python
import re

# A mockup of a SVG.
svg_body = "<svg>...</svg>"

# Keeps the amount of images/raster founded.
is_img = len(re.findall("=\"data:image\/.+;base64,", svg_body))

if is_img > 0:
  err_msg.append(f"- `{svg_path.name}` has an image/raster within it.")
```
> *This is my first time ever coding in that programming language.*

**Comment by Snailedlt** [comment]
@lunatic-fox no need to re-invent the wheel here. I think we can just re-use what SVGO has here: https://github.com/svg/svgo/blob/ae32acf5144be318872a45f49b3ece50c0a4cb18/plugins/removeRasterImages.js#L24

**Comment by lunatic-fox** [comment]
Hmm, the idea came from [here](https://github.com/devicons/devicon/pull/1554#discussion_r1030304788). They're regex is a step before and it will totally work.

The full regular expression with common image extensions would be something like this:
```regexp
/<image.+(xlink:)?href="data:image\/(png|jpg|jpeg|gif|tiff|svg\+xml|bmp);base64,.+"\/>/gm
```

But since there are more extensions we could reduce to this:
```regexp
/<image.+(xlink:)?href="data:image\/.+;base64,.+"\/>/gm
```

Or to simplify everything we can just ask if the svg has the `<image>` tag, that may be more than enough to know if there's a image/raster. So we can reuse [line 22](https://github.com/svg/svgo/blob/ae32acf5144be318872a45f49b3ece50c0a4cb18/plugins/removeRasterImages.js#L22) instead. 🙂

**Comment by Snailedlt** [comment]
Yeah, seems like a good idea to just generally not allow the image element. That way we disallow raster images and other linked images, so we have the entire source code in our repo instead of linking to external images.

**Comment by Snailedlt** [comment]
@lunatic-fox The code looks great! Nice work!

We should make the error message a big clearer though. 
Maybe like this?
`The icon "{svg_path.name}"  contains the <image> tag. This is usually a raster image, and therefore needs to be removed`
